modules/volunteers/sitesupport/hourTable.js

```diff
@@ -110,6 +36,90 @@ export default {
     }
   },
   methods: {
+    load() {
+      this.records = null;
+      this.totalHours = 0;
+      this.totalSpentHours = 0;
+      this.hours = [];
+      this.totalVolunteers = null;
+
+      showSpinner();
+      this.summeryColumns = [
+        {value:'name', title:'Department Worked', source:null},
+
+        {value:'entry_count', title:'Total Entries', source:this.hours},
+        {value:'volunteer_count', title:'Total Volunteers', source:this.hours},
+        {value:'total_hours', title:'Total Hours', source: this.hours}
+      ];
+      this.detailColumns = [
+        {value:'name', title:'Department Worked', source:null},
+        {value:'end', title:'Ended', source: this.hours},
+        {value:'hours', title:'Hours', source: this.hours}
+      ];
+
+      apiRequest('GET', '/department','max_results=all')
+        .then((response) => {
+          const result = JSON.parse(response.responseText);
+          this.departments = [];
+          result.data.forEach((entry) => {
+            this.departments.push(entry);
+          });
+
+          this.departments.sort((a,b) => (a.name > b.name) ? 1 : -1);
+
+          if (this.$parent.userId != null) {
+            this.columns = this.detailColumns;
+            this.records = this.hours;
+            apiRequest('GET', '/member/' + this.$parent.userId + '/volunteer/hours','max_results=all')
+              .then((response) => {
+                const result = JSON.parse(response.responseText);
+                result.data.forEach((entry) => {
+                  this.hours.push(entry);
+                })
+                this.hours.sort((a,b) => (a.department.name > b.department.name) ? 1 : -1);
+                this.totalSpentHours = 0;
+                this.totalHours = result.total_hours;
+                if (this.footer && this.$parent.userId != null) {
+                  apiRequest('GET', '/member/' + this.$parent.userId + '/volunteer/claims/summary','max_results=all')
+                    .then((response) => {
+                      const result = JSON.parse(response.responseText);
+                      this.totalSpentHours = result.spent_hours;
+                    })
+                    .finally(() => {
+                      hideSpinner();
+                      this.$emit('hourChange', this.totalHours, this.totalSpentHours);
+                    })
+                } else {
+                  hideSpinner();
+                  this.$emit('hourChange', this.totalHours, this.totalSpentHours);
+                }
+              })
+          } else {
+            this.columns = this.summeryColumns;
+            this.records = this.departments;
+            apiRequest('GET', '/event/current/volunteer/hours/summary','max_results=all')
+              .then((response) => {
+                const result = JSON.parse(response.responseText);
+                result.data.forEach((entry) => {
+                  this.hours[entry.department.id] = entry;
+                })
```

Comment:
Would this be cleaner?

`this.hours = result.data;`

It seems like we want to reset the data, is there any reason why this might not be a good approach?

Reply:
The hours table expected the index to be the department id.  and the data coming back from the API request is not that way.